support 0-dim arrays

alyst opened the pull request:
Currently, 0-dimensional arrays are not properly handled by the generic `_writejson()`.
The proposed specialized method just writes the first (and only) element.

TotalVerb commented:
Could you add a test?

TotalVerb commented:
It seems easier (and allocation-free) to write this as a `lower` method instead.

TotalVerb commented:
@alyst I see now why this might not work as a `lower` method, since `lower` wouldn't be called for any subtype of `AbstractArray`. Could you update this PR for the new `show_json` method?

TotalVerb commented:
In fact, the addition may be as simple as

```julia
function show_json{T,n}(io::SC, s::CS, A::AbstractArray{T,0})
    show_json(io, s, A[])
end
```

alyst commented:
@TotalVerb thanks for the suggestions. I've added the test and rebased the PR. I've also checked: if the `show_json(A::AbstractArray{T,0})` method is not defined, the new test is stuck at `newdims = ntuple(...)` of `show_json(A::AbstractArray{T,N})`. Would it make sense to add `@assert n > 0` there?

alyst commented:
travis failure seems unrelated
